corehq/apps/userreports/templates/userreports/edit_data_source.html

```diff
@@ -4,6 +4,80 @@
 {% load i18n %}
 
 {% block page_content %}
+  {% if data_source.get_id %}
+    <div class="btn-toolbar pull-right">
+        <div class="btn-group">
+          <a href="{% url 'preview_configurable_data_source' domain data_source.get_id %}" class="btn btn-default">{% trans 'Preview Data' %}</a>
+          <a href="{% url 'summary_configurable_data_source' domain data_source.get_id %}" class="btn btn-default">{% trans 'Data Source Summary' %}</a>
+        </div>
+        <div class="btn-group">
+          <form method='post' action="{% url 'rebuild_configurable_data_source' domain data_source.get_id %}" >
+            {% csrf_token %}
+            <div {% if data_source.disable_destructive_rebuild %}style="margin-right: 15px;"{% endif %}>
+              <button type="submit" class="btn btn-default disable-on-submit" {% if data_source.disable_destructive_rebuild %}disabled{% endif %}>
+                {% trans 'Rebuild Data Source'%}
+              </button>
+              {# janky: can't use a tooltip on a disabled button, so instead add a help icon and force extra space (above) so it's clear which button this is for #}
```

Comment:
can we then not use a tooltip and show a text note some other way?
I can't see in the images attached in the PR to see how this would look so i can't say how this is looking and if that is the best possible approach here.

Reply:
It looks like this:

<img width="1353" alt="Screen Shot 2019-10-22 at 4 59 56 PM" src="https://user-images.githubusercontent.com/1486591/67332726-6e82c080-f4ed-11e9-8691-6a9bd0974227.png">

This page used to hide the rebuild button and display a page-level alert at the bottom of the page. I removed this in https://github.com/dimagi/commcare-hq/pull/25662/commits/4d23681fefec254da94c5539a0caebe0a810ca4f because it was weirdly aligned. Moved to the top, it would look like this:

<img width="1348" alt="Screen Shot 2019-10-22 at 5 04 49 PM" src="https://user-images.githubusercontent.com/1486591/67333162-16988980-f4ee-11e9-87d9-da4cd2f8710e.png">

It's rather attention-grabbing. I think the icon is better UX, because only users looking at the rebuild button notice the message, instead of everyone who views the page.

Reply:
Thank you. Agreed on all of it. :)
A better and quick way for now can be to just have that button at the left most spot and the help note on its left. I am guessing keeping it at the right end would not be preferable because thats an ideal spot for the delete one. But considering this page is quite hidden, its fine to leave it as it is.

Reply:
> have that button at the left most spot and the help note on its left

But we should be consistent about where help icons go, and elsewhere in HQ they're on the left of the relevant element. There's no perfect answer here ;)